p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -440,46 +440,51 @@ public void onConnection(Connection connection) {
-                            if (connection.getPeersNodeAddressOptional().isPresent() &&
-                                    ownerNodeAddress.equals(connection.getPeersNodeAddressOptional().get())) {
-                                // We have a RequiresLiveOwnerData data object with the node address of the
-                                // disconnected peer. We remove that data from our map.
-
-                                // Check if we have the data (e.g. OfferPayload)
-                                ByteArray hashOfPayload = get32ByteHashAsByteArray(protectedStoragePayload);
-                                boolean containsKey = map.containsKey(hashOfPayload);
-                                if (containsKey) {
-                                    log.debug("We remove the data as the data owner got disconnected with " +
-                                            "closeConnectionReason=" + closeConnectionReason);
-
-                                    // We only set the data back by half of the TTL and remove the data only if is has
-                                    // expired after that back dating.
-                                    // We might get connection drops which are not caused by the node going offline, so
-                                    // we give more tolerance with that approach, giving the node the change to
-                                    // refresh the TTL with a refresh message.
-                                    // We observed those issues during stress tests, but it might have been caused by the
-                                    // test set up (many nodes/connections over 1 router)
-                                    // TODO investigate what causes the disconnections.
-                                    // Usually the are: SOCKET_TIMEOUT ,TERMINATED (EOFException)
-                                    protectedStorageEntry.backDate();
-                                    if (protectedStorageEntry.isExpired(this.clock)) {
-                                        log.info("We found an expired data entry which we have already back dated. " +
-                                                "We remove the protectedStoragePayload:\n\t" + Utilities.toTruncatedString(protectedStorageEntry.getProtectedStoragePayload(), 100));
-                                        removeFromMapAndDataStore(protectedStorageEntry, hashOfPayload);
-                                    }
-                                } else {
-                                    log.debug("Remove data ignored as we don't have an entry for that data.");
-                                }
-                            }
-                        }
-                    });
-        }
+        if (closeConnectionReason.isIntended)
+            return;
+
+        if (!connection.getPeersNodeAddressOptional().isPresent())
```

Comment:
I wish there was a better reason than this one, but I used this pattern because it reduces the number of lint errors from IDEA. If you don't have an `isPresent()` call prior to a `get()` it will highlight the word and raise an alert. You can suppress it, but that is just more code to add.

The other benefit is that mocking is bit less error-prone because you just need to mock out the Optional and not the wrapper class that has the `hasPeerNodeAddress()` function.